packages/sdk/src/syncedStream.ts

```diff
@@ -57,60 +140,126 @@ export class SyncedStream extends Stream implements ISyncedStream {
+                minipoolEvents: events,
+                lastMiniblockNum: miniblocks[miniblocks.length - 1].header.miniblockNum,
+            })
+
+            await this.persistenceStore.saveSyncedStream(this.streamId, cachedSyncedStream)
+            await this.persistenceStore.saveMiniblocks(this.streamId, miniblocks, 'forward')
+
+            this.metrics.initializationTime = performance.now() - startTime
+            this.metrics.memoryUsage = this.calculateMemoryUsage()
+            this.metrics.successfulOperations++
+
+            if (this.metrics.memoryUsage > MEMORY_THRESHOLD) {
+                this.log('High memory usage detected', {
+                    streamId: this.streamId,
+                    memoryUsage: this.metrics.memoryUsage,
+                    threshold: MEMORY_THRESHOLD,
+                })
+            }
+
+            this.markUpToDate()
+        } catch (error) {
+            this.metrics.failedOperations++
+            this.metrics.lastError = error instanceof Error ? error : new Error(String(error))
+            this.logError('Error in stream initialization', {
+                streamId: this.streamId,
+                error: this.metrics.lastError.message,
+            })
+            throw error
+        }
     }
 
     async initializeFromResponse(response: ParsedStreamResponse) {
-        this.log('initializing from response', this.streamId)
-        const cleartexts = await this.persistenceStore.getCleartexts(response.eventIds)
-        await this.initialize(
-            response.streamAndCookie.nextSyncCookie,
-            response.streamAndCookie.events,
-            response.snapshot,
-            response.streamAndCookie.miniblocks,
-            [],
-            response.prevSnapshotMiniblockNum,
-            cleartexts,
-        )
-        this.markUpToDate()
+        const startTime = performance.now()
+        this.metrics.lastAccessTime = Date.now()
+
+        try {
+            this.log('initializing from response', this.streamId)
+            const cleartexts = await this.persistenceStore.getCleartexts(response.eventIds)
+
+            await this.initialize(
+                response.streamAndCookie.nextSyncCookie,
+                response.streamAndCookie.events,
+                response.snapshot,
+                response.streamAndCookie.miniblocks,
+                [],
+                response.prevSnapshotMiniblockNum,
+                cleartexts,
+            )
+
+            this.metrics.initializationTime = performance.now() - startTime
+            this.metrics.successfulOperations++
+            this.markUpToDate()
+        } catch (error) {
+            this.metrics.failedOperations++
+            this.metrics.lastError = error instanceof Error ? error : new Error(String(error))
+            this.logError('Error initializing from response', {
+                streamId: this.streamId,
+                error: this.metrics.lastError.message,
+            })
+            throw error
```

Comment:
its nice to track this time... but we're tracking it already in the syncedStreamsExtensions